src/dotnet-install.sh

```diff
@@ -1134,7 +1175,7 @@ if [ "$dry_run" = true ]; then
     if [ "$valid_legacy_download_link" = true ]; then
         say "Legacy named payload URL: $legacy_download_link"
     fi
-    repeatable_command="./$script_name --version "\""$specific_version"\"" --install-dir "\""$install_root"\"" --architecture "\""$normalized_architecture"\"""
+    repeatable_command="./$script_name --version "\""$specific_version"\"" --install-dir "\""$install_root"\"" --architecture "\""$normalized_architecture"\"" --os "\""$normalized_os"\""" 
```

Comment:
The `--os` option does not work for legacy links for some OS. So, by using this command you run the scripts which throw a warning. And if it would be a terminating error, that it will be the wrong command entirely (+one small argument for a warning rather than error, I see). 
One option to avoid it is to add `normalized_os` only when the user specifies `user_defined_os`.

Reply:
`--os` option is not used when creating legacy link, so it won't make any change if included to the command. 
Using it for legacy link will not give warning or error.